ddocs/erlang_filters/filters/doc_by_place.erl

```diff
@@ -0,0 +1,98 @@
+
+% Made "cmd+v futon-ready" with the following disgusting display of black magic
+% cat scratch.erl | sed 's/%.*//' | sed 's/"/\\"/'g | sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/ /g' | pbcopy
+fun ({Doc}, {Req}) ->
+
+  % Some values are wrapped in a tuple for some arbitrary reason
+  %  i.e. {[{...}]} instead of [{...}]
+  % Instead of working this out we're just guarding against it
+  % TODO work out all the rules for this so we can optimise this function
+  %      away for better performance
+  SafeGetValue = fun(Key, PropListMaybe, Default) ->
+    if
+      is_tuple(PropListMaybe) andalso size(PropListMaybe) == 1 ->
+        proplists:get_value(Key, element(1, PropListMaybe), Default);
+      is_list(PropListMaybe) ->
+        proplists:get_value(Key, PropListMaybe, Default);
+      true ->
+        Default
+    end
+  end,
+
+  DeepGet = fun(DeepGet, KeyList, PropList) ->
+    if
+      length(KeyList) > 1 andalso PropList == [] ->
+        undefined;
+      length(KeyList) == 1 ->
+        SafeGetValue(hd(KeyList), PropList, undefined);
+      true ->
+        DeepGet(DeepGet, tl(KeyList), SafeGetValue(hd(KeyList), PropList, []))
+    end
+  end,
+
+  FindPlace = fun(FindPlace, Place, QueryId) ->
+    if
+      Place == undefined ->
+        false;
+      true ->
+        case SafeGetValue(<<"_id">>, Place, undefined) =:= QueryId of
+          true -> true;
+          false -> FindPlace(FindPlace, SafeGetValue(<<"parent">>, Place, undefined), QueryId)
+        end
+    end
+  end,
+
+  Ok = fun(Place) ->
+    QueryId = DeepGet(DeepGet, [<<"query">>, <<"id">>], Req),
+    case QueryId =:= undefined of
+      true -> true; % Admin
+      false ->
+        case Place =:= undefined of
+          true -> DeepGet(DeepGet, [<<"query">>, <<"unassigned">>], Req) =:= <<"true">>;
+          false -> FindPlace(FindPlace, Place, QueryId)
+        end
+    end
+  end,
+
+  CouchUser = list_to_binary([
+      <<"org.couchdb.user:">>,
+      DeepGet(DeepGet, [<<"userCtx">>, <<"name">>], Req)]),
+
+  case proplists:get_value(<<"_id">>, Doc) of
+    <<"design/medic">> -> false;     % Never replicate the ddoc as it's too big.
+                                     % Changes are tracked in the watchDesignDoc function of the DB service.
+    <<"resources">> -> true;
+    CouchUser -> true;
+    _ -> case proplists:get_value(<<"type">>, Doc) of
+          <<"data_record">> ->
+            case proplists:get_value(<<"kujua_message">>, Doc) of
+              true ->
+                Tasks = proplists:get_value(<<"tasks">>, Doc, []),
+                if
+                  length(Tasks) > 0 ->
+                    % Outgoing message
+                    Ok(
+                      proplists:get_value(<<"contact">>,
+                        element(1, hd(proplists:get_value(<<"messages">>,
+                          element(1, hd(Tasks)))))));
+                  true ->
+                    false
```

Comment:
If `Tasks.length == 0`, this returns false.
The js version returns `ok(false)`, which is `req.query.unassigned === 'true'`
https://github.com/medic/medic-webapp/blob/9da1a7efedcc4d6295390d1162d314c4e6f07c54/lib/filters.js#L23